server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -272,7 +272,11 @@ public void openConnection(DiscoveryNode node, ConnectionProfile profile, Action
             throw new ConnectTransportException(null, "can't open connection to a null node");
         }
         ConnectionProfile finalProfile = maybeOverrideConnectionProfile(profile);
-        closeLock.readLock().lock(); // ensure we don't open connections while we are closing
+        if (closeLock.readLock().tryLock() == false) {
```

Comment:
Do we need a timeout here or is it okay to fail immediately if we can't acquire the lock?

Reply:
No we really don't want to ever block here as this may be called on a transport thread and we only expect not to be able to acquire the lock while the transport is shutting down as well.

Reply:
Ah, got it. Because it's the transport thread, we would rather fail immediately and free up resources rather than try to make progress :+1: